core/src/main/scala/org/apache/spark/api/r/SerDe.scala

```diff
@@ -166,6 +166,24 @@ private[spark] object SerDe {
     }
   }
 
+  def readEnv(in: DataInputStream): java.util.Map[Object, Object] = {
```

Comment:
I just want to keep readMap() for the case where in R side an environment in which all values are of same type, the value type is serialized once. It is optimized than readEnv(). But I am OK if you think we can update readMap() and remove readEnv() for now and do such optimization in the future if necessary.